projects/plugins/jetpack/extensions/shared/components/product-management-controls/use-products.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { jetpackMembershipProductsStore } from './store';
+
+export default function useProducts( {
+	selectedProductId = 0,
+	setSelectedProductId = () => {},
+	shouldSkipResolver = false,
+} ) {
```

Comment:
What's the benefit of using an object as an argument instead of actually using a list of arguments?

Reply:
Mostly personal preference, as I find it more readable and predictable.
I'm definitely not a fan of having flags (such as `shouldSkipResolver`) as regular arguments, as they become easily confusing.

For example, here it would be used like this:
```js
useProducts( selectedPlanId, setSelectedProductId, isPreview )
```
That `isPreview` is not immediately clear what it means, if it comes from the hook or from the consumer.